File: Python/8py 3/shorten_to_date.py
"""
|--------------------------------------------------------------------------
| Remove the time
|--------------------------------------------------------------------------
|
"""

def shorten_to_date(long_date):
   return long_date.split(',')[0]


def shorten_to_date(long_date):
   return long_date[:long_date.index(',')]


def shorten_to_date(long_date):
   return long_date[:long_date.find(',')]


print(shorten_to_date("Monday February 2, 8pm"), "Monday February 2")
print(shorten_to_date("Tuesday May 29, 8pm"), "Tuesday May 29")
print(shorten_to_date("Wed September 1, 3am"), "Wed September 1")
print(shorten_to_date("Friday May 2, 9am"), "Friday May 2")
print(shorten_to_date("Tuesday January 29, 10pm"), "Tuesday January 29")

File: PHP/8php 3/shortenToDate.php
<?php
/*
|--------------------------------------------------------------------------
| Remove the time
|--------------------------------------------------------------------------
|
*/

function shortenToDate($longDate) {
   return explode(',', $longDate)[0];
}


function shortenToDate($str) {
   return preg_replace('/,.*$/', '', $str);
}


function shortenToDate($longDate) {
   return substr($longDate, 0, strpos($longDate, ','));
}


function shortenToDate($d) {
   return preg_replace('/\, \d+(a|p)m/', '', $d);
}


print_r(shortenToDate("Friday May 2, 9am") . " Friday May 2
");
print_r(shortenToDate("Tuesday January 29, 10pm") . " Tuesday January 29
");
print_r(shortenToDate("Monday December 25, 10pm") . " Monday December 25
");